components/layout/TopNav.tsx

```
@@ -0,0 +1,144 @@
+      </Link>
+      <div className="hidden md:flex items-center gap-2 flex-1 max-w-md">
+        <div className="relative flex-1">
+          <input
+            value={viewId}
+            onChange={e=>setViewId(e.target.value)}
+            onKeyDown={e=>{ if(e.key==='Enter'){ goView() } }}
+            placeholder="View any shared wallet (name/id)"
+            className="w-full h-9 rounded-lg input-glass px-3 pr-9 text-sm font-mono"
+            aria-label="View any wallet by name or id"
+          />
+          <button aria-label="Open wallet" onClick={goView} className="absolute right-1.5 top-1.5 h-6 w-6 rounded-md bg-white/10 hover:bg-white/15 flex items-center justify-center"><Search className="h-3.5 w-3.5" /></button>
```

Biome (2.1.2)

[error] Provide an explicit type prop for the button element.

The default type of a button is submit, which causes the submission of a form when placed inside a `form` element. This is likely not the behaviour that you want inside a React application.
Allowed button types are: submit, button or reset

(lint/a11y/useButtonType)